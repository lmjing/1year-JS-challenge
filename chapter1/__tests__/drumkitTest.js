document.body.innerHTML = `<div id="content"></div>`;

test('test snapshots - contents result', function () {
    let { contents, data} = require("../src/controller");

    let content = document.getElementById("content");
    content.innerHTML = contents(data);
    expect(document.body).toMatchSnapshot();
});

test("test snapshots - test", () => {
    expect('<ul><li><a href="#">My HTML</a></li></ul>').toMatchSnapshot();
});

// test("test keydown event", () => {
//     import { setKeyboardEvent } from "../src/controller";
// });
