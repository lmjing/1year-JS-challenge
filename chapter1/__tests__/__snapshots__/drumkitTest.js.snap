// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshots - contents result 1`] = `
<body>
    <div id="content">
        <div class="box" data-key="A">
            <kbd class="box--key">A</kbd>
            <div class="box--sound">clap</div>
        </div>

        <div class="box" data-key="S">
            <kbd class="box--key">S</kbd>
            <div class="box--sound">hihat</div>
        </div>

        <div class="box" data-key="D">
            <kbd class="box--key">D</kbd>
            <div class="box--sound">kick</div>
        </div>

        <div class="box" data-key="F">
            <kbd class="box--key">F</kbd>
            <div class="box--sound">openhat</div>
        </div>

        <div class="box" data-key="G">
            <kbd class="box--key">G</kbd>
            <div class="box--sound">boom</div>
        </div>

        <div class="box" data-key="H">
            <kbd class="box--key">H</kbd>
            <div class="box--sound">ride</div>
        </div>

        <div class="box" data-key="J">
            <kbd class="box--key">J</kbd>
            <div class="box--sound">snare</div>
        </div>

        <div class="box" data-key="K">
            <kbd class="box--key">K</kbd>
            <div class="box--sound">tom</div>
        </div>

        <div class="box" data-key="L">
            <kbd class="box--key">L</kbd>
            <div class="box--sound">tink</div>
        </div>
        </div>
    <script type="application/javascript" src="controller.js"></script>


</body>`;

exports[`test snapshots - test 1`] = `
<ul>
  <li>
    <a href="#">
      My HTML
    </a>
  </li>
</ul>`;
